ts/main.ts

```diff
@@ -46,3 +130,166 @@ function testKey(key: string): boolean {
 function testKeyStrict(key: string): boolean {
   return /^[A-Za-z0-9 ]$/.test(key);
 }
+
+function getRequestUrl(
+  option: string,
+  keyword: string,
+  url: string = 'https://api.datamuse.com/words?',
+): string {
+  //  Autocomplete suggestions only
+  if (option.includes('autocomplete')) {
+    url = `https://api.datamuse.com/sug?s=${keyword}`;
+  } else {
+    if (url !== 'https://api.datamuse.com/words?') {
+      if (url.includes('?')) {
+        if (!url.endsWith('&')) {
+          url += '&';
+        }
+      }
+    }
+    //  Root options
+    if (option.includes('meansLike')) {
+      url += `ml=${keyword}&`;
+    }
+    if (option.includes('soundsLike')) {
+      url += `sl=${keyword}&`;
+    }
+    if (option.includes('spelledLike')) {
+      url += `sp=${keyword}&`;
+    }
+    if (option.includes('followsWords')) {
+      url += `lc=${keyword}&`;
+    }
+    if (option.includes('followedWith')) {
+      url += `rc=${keyword}&`;
+    }
+    if (option.includes('startsWith')) {
+      url += `sp=${keyword}*&`;
+    }
+    if (option.includes('endsWith')) {
+      url += `sp=*${keyword}&`;
+    }
+    //  Related word options
+    if (option.includes('describesWords')) {
+      url += `rel_jja=${keyword}&`;
+    }
+    if (option.includes('describedWith')) {
+      url += `rel_jjb=${keyword}&`;
+    }
+    if (option.includes('associatedWith')) {
+      url += `rel_trg=${keyword}&`;
+    }
+  }
+  //  Trim last &amp;
+  if (url.endsWith('&')) {
+    if (url.includes('max')) {
+      url = url.substring(0, url.length - 1);
+    } else {
+      url += 'max=10';
+    }
+  }
+  return url;
+}
```

Comment:
We can simplify this by using a map to associate options with their corresponding URL parameters. This way we can reduce the amount of `if` statements that would be necessary. 

```typescript
function getRequestUrl(
  option: string,
  keyword: string,
  url: string = 'https://api.datamuse.com/words?',
): string {
  const optionsMap = {
    autocomplete: `https://api.datamuse.com/sug?s=${keyword}`,
    meansLike: 'ml',
    soundsLike: 'sl',
    spelledLike: 'sp',
    followsWords: 'lc',
    followedWith: 'rc',
    startsWith: 'sp',
    endsWith: 'sp',
    describesWords: 'rel_jja',
    describedWith: 'rel_jjb',
    associatedWith: 'rel_trg',
  };

  if (option in optionsMap) {
    if (option === 'autocomplete') {
      return optionsMap[option];
    } else {
      url += `${optionsMap[option]}=${keyword}&`;
    }
  }

  if (url.endsWith('&')) {
    url = url.includes('max') ? url.slice(0, -1) : url + 'max=10';
  }

  return url;
}
```